openedx/core/djangoapps/user_authn/views/auto_auth.py

```diff
@@ -170,9 +170,9 @@ def auto_auth(request):  # pylint: disable=too-many-statements
         elif course_id:
             # Redirect to the course homepage (in LMS) or outline page (in Studio)
             try:
```

Comment:
Just checking - IS the logic reversed?

Reply:
Yes I should have commented instead of just giving a thumbs up  :)

I did reverse the logic. The old code was using url reverse-name lookups as a trick to detect whether we're in LMS or Studio side. (Feels like a weird trick to do that instead of something more explicit - I've seen checks like `ROOT_URLCONF == 'lms.urls'` as another way. Feels like we need a `am_in_lms()` method or something.)

But because I refactored `course_home_url_name` to return a url instead of a url-name (and renamed it to `course_home_url`), I just switched which url-name we were looking up first, to detect lms or studio.

`course_handler` is only defined on the Studio side, so if that `reverse` lookup works, we're in Studio. If it doesn't, we grab the course home URL for the LMS side.

Reply:
Thanks. That WAS confusing. Thanks for unwinding it.